schema.yaml

```diff
@@ -0,0 +1,364 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+type: object
+
+# Every document requires a version
+properties:
+  version:
+    $ref: "#/$defs/XtpVersion"
+required:
+  - version
+
+allOf:
+  # In v0 we only support simple named exports
+  - if:
+      properties:
+        version:
+          const: v0
+    then:
+      properties:
+        exports:
+          type: array
+          items:
+            type: string
+            pattern: "^[a-zA-Z_$][a-zA-Z0-9_$]*$"
+        version:
+          const: v0
+      required:
+        - exports
+      additionalProperties: false
+  # In v1 we support complex exports, imports, and components
+  - if:
+      properties:
+        version:
+          const: v1-draft
+    then:
+      properties:
+        version:
+          $ref: "#/$defs/XtpVersion"
+        exports:
+          type: object
+          patternProperties:
+            "^[a-zA-Z_$][a-zA-Z0-9_$]*$":
+              $ref: "#/$defs/Export"
+          additionalProperties: false
+        imports:
+          type: object
+          patternProperties:
+            "^[a-zA-Z_$][a-zA-Z0-9_$]*$":
+              $ref: "#/$defs/Import"
+          additionalProperties: false
+        components:
+          type: object
+          properties:
+            schemas:
+              type: object
+              patternProperties:
+                "^[a-zA-Z_$][a-zA-Z0-9_$]*$":
+                  $ref: "#/$defs/Schema"
+              additionalProperties: false
+          required:
+            - schemas
+          additionalProperties: false
+      required:
+        - exports
+      additionalProperties: false
+$defs:
```

Comment:
@mhmd-azeez I've made a lot of simplifications to the json schema. I suspect we can go further but not 100% sure yet. Curious if you could take a look.